### PR TITLE
LPD-23926 Apply the new configuration to allow or prevent the inclusion of the parent sites in the notification recipient, when it type is site role

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPD-23926

### CODE REVIEW

**Comment by p-albuquerque** [comment]
Depends on: https://github.com/brianchandotcom/liferay-portal/pull/149628

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]
Hey @linolaoj I've reviewed this pull request, @p-albuquerque please see my comments.

Thanks!

File: modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/notification/recipient/ScriptNotificationRecipientBuilder.java [hunk 1]
```diff
@@ -94,7 +94,9 @@ public void processKaleoTaskAssignmentInstance(
 
 	@Activate
 	@Override
-	protected void activate(BundleContext bundleContext) {
+	protected void activate(
+		BundleContext bundleContext, Map<String, Object> properties) {
```
[the remark this answers is not in the document]
Comment (p-albuquerque) [reply modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/notification/recipient/ScriptNotificationRecipientBuilder.java:98]: Hi @MarinhoFeliphe  If I remove `Map<String, Object> properties` parameter here, I'm forced to remove the `@Override`, so it means that [ScriptNotificationRecipientBuilder.java](https://github.com/liferay-workflow/liferay-portal/pull/1134/files/24ac3e4619271edafddf895ef3fcaa99db3c1854#diff-b6a07d9633d7c0900201d182e6d3b9c2815cb8614e1c1f9364761c852e2480a3) will contains two activate methods. It's okay keep this way?

File: modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/notification/recipient/RoleNotificationRecipientBuilder.java [hunk 2]
```diff
@@ -85,9 +90,13 @@ public void processKaleoTaskAssignmentInstance(
...
 
 	@Activate
-	protected void activate(BundleContext bundleContext) {
+	protected void activate(
+		BundleContext bundleContext, Map<String, Object> properties) {
+
 		_serviceTrackerList = ServiceTrackerListFactory.open(
 			bundleContext, GroupAwareRoleValidator.class);
+		_workflowDefinitionConfiguration = ConfigurableUtil.createConfigurable(
+			WorkflowDefinitionConfiguration.class, properties);
```
[the remark this answers is not in the document]
Comment (p-albuquerque) [reply modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/notification/recipient/RoleNotificationRecipientBuilder.java:99]: Hi @MarinhoFeliphe even without the `@Modified` annotation, it have been worked. I change the configuration several times and I was faced no issues. But I will try it with this annotation as well

File: modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowTaskManagerImplTest.java [hunk 3]
```diff
@@ -663,32 +669,88 @@ public void testApproveOrganizationParentReviewerWithoutSite()
...
+
+		_assignWorkflowTaskToUser(_adminUser, childSiteMemberUser);
+
+		_checkUserNotificationEventsByUsers(1, childSiteMemberUser);
+
+		_completeWorkflowTask(childSiteMemberUser, Constants.APPROVE);
+
+		_getWorkflowInstance(JournalArticle.class.getName(), article.getId());
+
+		article = _journalArticleLocalService.getArticle(article.getId());
```
[the remark this answers is not in the document]
Comment (p-albuquerque) [reply modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowTaskManagerImplTest.java:752]: Hi @MarinhoFeliphe I will put them in another test method, then I was thinking about use another asset to test it like `BlogsEntry`. What do you think?

File: modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowTaskManagerImplTest.java [hunk 4]
```diff
@@ -176,6 +178,8 @@ public static void setUpClass() throws Exception {
 			_configuration,
 			HashMapDictionaryBuilder.<String, Object>put(
 				"company.administrator.can.publish", true
+			).put(
+				"preventNotifyingAncestorSites", false
```
[the remark this answers is not in the document]
Comment (p-albuquerque) [reply modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowTaskManagerImplTest.java:182]: I was removed it and tested again. Everything works well! 
Thanks

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

**Comment by p-albuquerque** [comment]
ci:test:workflow

**Comment by p-albuquerque** [comment]
ci:test:workflow-metrics

**Pull request closed without merging**